src/jump_moi_overloads.jl

```diff
@@ -1,6 +1,127 @@
+end
+function MOIU.isapprox_zero(func::RankOneQuadraticFunction, tol)
+    return MOIU.isapprox_zero(standard_form(func), tol)
+end
+
+struct IndexMappedFunction{F<:MOI.AbstractScalarFunction} <: AbstractLazyScalarFunction
+    func::F
+    index_map::MOIU.IndexMap
+end
+function JuMP.coefficient(func::IndexMappedFunction, vi::MOI.VariableIndex)
+    return JuMP.coefficient(func.func, func.index_map[vi])
+end
+function quad_sym_half(func::IndexMappedFunction, vi1::MOI.VariableIndex, vi2::MOI.VariableIndex)
+    return quad_sym_half(func.func, func.index_map[vi1], func.index_map[vi2])
+end
+function JuMP.coefficient(func::IndexMappedFunction, vi1::MOI.VariableIndex, vi2::MOI.VariableIndex)
+    return JuMP.coefficient(func.func, func.index_map[vi1], func.index_map[vi2])
+end
+function standard_form(func::IndexMappedFunction)
+    return MOIU.map_indices(func.index_map, standard_form(func.func))
+end
+MOIU.isapprox_zero(func::IndexMappedFunction, tol) = MOIU.isapprox_zero(func.func, tol)
+
+function MOIU.map_indices(index_map::MOIU.IndexMap, func::AbstractLazyScalarFunction)
+    return IndexMappedFunction(func, index_map)
+end
+
+struct MOItoJuMP{F<:MOI.AbstractScalarFunction} <: JuMP.AbstractJuMPScalar
+    model::JuMP.Model
+    func::F
+end
+Base.broadcastable(func::MOItoJuMP) = Ref(func)
+function JuMP.coefficient(func::MOItoJuMP, var_ref::JuMP.VariableRef)
+    check_belongs_to_model(var_ref, func.model)
+    return JuMP.coefficient(func.func, JuMP.index(var_ref))
+end
+function quad_sym_half(func::MOItoJuMP, var1_ref::JuMP.VariableRef, var2_ref::JuMP.VariableRef)
+    check_belongs_to_model(var1_ref, func.model)
+    return quad_sym_half(func.func, JuMP.index(vi1), JuMP.index(var2_ref))
+end
+function JuMP.coefficient(func::MOItoJuMP, var1_ref::JuMP.VariableRef, var2_ref::JuMP.VariableRef)
+    check_belongs_to_model(var2_ref, func.model)
+    return JuMP.coefficient(func.func, JuMP.index(vi1), JuMP.index(var2_ref))
+end
+function Base.convert(::Type{JuMP.GenericAffExpr{T,JuMP.VariableRef}}, func::MOItoJuMP) where {T}
+    return JuMP.GenericAffExpr{T,JuMP.VariableRef}(func.model, convert(MOI.ScalarAffineFunction{T}, func.func))
+end
+function Base.convert(::Type{JuMP.GenericQuadExpr{T,JuMP.VariableRef}}, func::MOItoJuMP) where {T}
+    return JuMP.GenericQuadExpr{T,JuMP.VariableRef}(func.model, convert(MOI.ScalarQuadraticFunction{T}, func.func))
+end
+JuMP.moi_function(func::MOItoJuMP) = func.func
+function JuMP.jump_function(model::JuMP.Model, func::AbstractLazyScalarFunction)
+    return MOItoJuMP(model, func)
+end
+function standard_form(func::MOItoJuMP)
+    return JuMP.jump_function(func.model, standard_form(func.func))
+end
+function JuMP.function_string(mode, func::MOItoJuMP)
+    return JuMP.function_string(mode, standard_form(func))
+end
+
+function MOI.get(model::JuMP.Model, attr::BackwardOutObjective)
+    func = MOI.get(JuMP.backend(model), attr)
+    return JuMP.jump_function(model, func)
```

Comment:
do we still need the function below this one?

Reply:
Yes, we still have BackwardIn and BackwardOut for the constraints.